src/react/components/shell/preditcPageRoute.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { useEffect, useState } from "react";
+import { Route } from "react-router-dom";
+import { useSelector } from 'react-redux'
+
+import PredictPage from "../pages/predict/predictPage";
+import { IApplicationState } from '../../../models/applicationState';
+
+
+/**
+ * @name - Predict page Route
+ * @description - Controls rendering of predict page
+ */
+
+export function PredictPageRoute() {
+    const projectProperties = useSelector((state: IApplicationState) => {
+        if (state && state.currentProject) {
+            const { apiKey, folderPath, apiUriBase, id, trainRecord } = state.currentProject;
+            let modelId: string;
+            if (trainRecord) {
+                modelId = trainRecord.modelInfo.modelId;
+            }
+            return JSON.stringify({ id, apiKey, apiUriBase, folderPath, modelId });
```

Comment:
could leave out apiKey and apiUriBase I think

Reply:
I thought if they will switch, lets say, different version on the service - result can varies so we should unmount prediction...